app/src/main/java/org/mozilla/fenix/utils/Undo.kt

```diff
@@ -42,43 +47,55 @@ fun CoroutineScope.allowUndo(
     // writing a volatile variable.
     val requestedUndo = AtomicBoolean(false)
 
-    // CONFIRMATION DIALOG: TO-D0 CHECK IF THERE IS TALK BACK VIA ACCESSIBILITY THEN DO THIS
-    val dialogBuilder = AlertDialog.Builder(view.context)
-    dialogBuilder.setMessage(message).setCancelable(false).setPositiveButton("Yes"){_, _ ->
-        launch {
-            onCancel.invoke()
-        }
-    }
-
-    // create dialog box
-    val alert = dialogBuilder.create()
-    alert.setTitle("UNDO")
-    alert.show()
+    val am = view.context.getSystemService(Context.ACCESSIBILITY_SERVICE) as AccessibilityManager
+    val isAccessibilityEnabled = am.isEnabled
 
-
-    // Launch an indefinite snackbar.
-    val snackbar = FenixSnackbar
-        .make(view, FenixSnackbar.LENGTH_INDEFINITE)
-        .setText(message)
-        .setAnchorView(anchorView)
-        .setAction(undoActionTitle) {
-            requestedUndo.set(true)
+    if (isAccessibilityEnabled) {
+        // CONFIRMATION DIALOG: TO-D0 CHECK IF THERE IS TALK BACK VIA ACCESSIBILITY THEN DO THIS
```

Comment:
The code looks good, but these blocks have gotten big enough that they're getting a little tough to follow.  I think this would be easier if they were broken into helper functions, so we could get:
```kotlin
if (isAccessibilityEnabled) {
  showUndoDialog()
else {
  showUndoSnackbar()
}
```